core/arch/arm/mm/core_mmu.c

```diff
@@ -1595,21 +1595,22 @@ static void *phys_to_virt_tee_ram(paddr_t pa)
 	uint32_t a;
 	paddr_t p;
 
-	if (pa >= CFG_TEE_LOAD_ADDR && pa < get_linear_map_end())
+	if (pa >= CFG_TEE_LOAD_ADDR &&
+	    pa < virt_to_phys((void *)get_linear_map_end()))
```

Comment:
Looks strange to call `virt_to_phys()` for a known linear address.
I would recommend to discard this change and introduce it back once we don't require the linear mapping.